### PR TITLE
docs(router): clarify dynamic route slug is about a file name, not "folder"

### PR DESCRIPTION
**Pull request by cmbritten**
### What?

Update documentation on dynamic routing to clarify the use of segment names within file names, rather than referring to them as "folder names".

### Why?

The original documentation used the example term `folderName` when it was referencing dynamic segment names within filenames.  The paths being described here are file names but not folders, and end in a file extension.  This could confuse readers and developers, especially those new to Next.js, leading to potential misunderstandings and misconfigurations.  Perhaps folderName was a relic of a different example wrapping a folder name to do a nested path or other configuration?  But here it seems out of place.

### How?

- Reviewed the provided documentation sections on dynamic routing.
- Replaced references to "folderName" with a different term "segmentName". 
- Ensured the examples provided in the documentation correctly showcase the use of segment names within filenames.

### CODE REVIEW

File: docs/03-pages/01-building-your-application/01-routing/02-dynamic-routes.mdx [hunk 1]
```diff
@@ -13,7 +13,7 @@ When you don't know the exact segment names ahead of time and want to create rou
 
 ## Convention
 
-A Dynamic Segment can be created by wrapping a folder's name in square brackets: `[folderName]`. For example, `[id]` or `[slug]`.
+A Dynamic Segment can be created by wrapping a filename (but not the extension) in square brackets: `[segmentName]`. For example, `[id]` or `[slug]`.
```
Comment (leerob) [inline docs/03-pages/01-building-your-application/01-routing/02-dynamic-routes.mdx:16]:
```suggestion
A Dynamic Segment can be created by wrapping a file or folder name in square brackets: `[segmentName]`. For example, `[id]` or `[slug]`.
```

Comment (cmbritten) [reply docs/03-pages/01-building-your-application/01-routing/02-dynamic-routes.mdx]: Thanks @leerob for the review and suggestion.  I will add that I don't recall seeing an example on this page of the folder name path pattern.  Maybe that would be a good addition as well, if that feature is still intended to be used (or perhaps I've missed it elsewhere).  Thanks again!

**Review by leerob — DISMISSED** [review]

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 6ebcc54e562edb550640453625e7cbd1f5b724c6

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 6ebcc54e562edb550640453625e7cbd1f5b724c6

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: 6ebcc54e562edb550640453625e7cbd1f5b724c6

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by leerob — APPROVED** [review]

**Pull request merged**